### PR TITLE
add srgan

### PR DESCRIPTION
**Pull request by yayeoCddy**
数据集和训练模型目前暂时放在oneflow-static.oss-cn-beijing.aliyuncs上

### CODE REVIEW

File: SRGAN/models/of_model.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        self.pixel_shuffle = nn.PixelShuffle(up_scale)
+        self.prelu = nn.PReLU()
+
+    def forward(self, x):
+        x = self.conv(x)
+        x = self.pixel_shuffle(x)
+        x = self.prelu(x)
+        return x
+
+
```
Comment (Ldpe2G) [inline SRGAN/models/of_model.py:114]: 以下的main函数部分就不用保留了

File: SRGAN/train_of_srgan.py [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+        hr_size=HR_SIZE,
+        lr_size=lr_size,
+        batch_size=opt.batch_size,
+    )
+    val_data = ValDatasetFromFolder(opt.data_dir, mode=modes[0], upscale_factor=4)
+    start_t = time.time()
+
+    netG = Generator(UPSCALE_FACTOR)
+    print("# generator parameters:", sum(param.numel() for param in netG.parameters()))
+    # netD = Discriminator(HR_SIZE)
```
Comment (Ldpe2G) [inline SRGAN/train_of_srgan.py:105]: 无用的注释代码都去掉

**Comment by Ldpe2G** [comment]
提供一下 requirements.txt 文件

File: SRGAN/train_of_srgan.py [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+            # d_loss = 1 - real_out + fake_out
+            d_loss.backward(retain_graph=True)
+            optimizerD.step()
+            optimizerD.zero_grad()
+
+            ############################
+            # (2) Update G network: minimize 1-D(G(z)) + Perception Loss + Image Loss + TV Loss
+            ###########################
+
+            g_loss = generator_criterion(fake_out, fake_img, real_img)
```
Comment (Ldpe2G) [inline SRGAN/train_of_srgan.py:185]: 你这里 fake_out 为什么不是等 判别器更新之后，再重新 fake_out = netD(fake_img)，之前的 fake_out 是
`fake_out = netD(fake_img.detach())` 这样子判别器的梯度传不回生成器？

File: SRGAN/models/of_model.py (hunk 1 above)
Comment (yayeoCddy) [reply SRGAN/models/of_model.py:114]: okok

File: SRGAN/train_of_srgan.py (hunk 2 above)
Comment (yayeoCddy) [reply SRGAN/train_of_srgan.py:105]: okok

File: SRGAN/models/of_model.py (hunk 1 above)
Comment (yayeoCddy) [reply SRGAN/models/of_model.py:114]: okok

File: SRGAN/train_of_srgan.py (hunk 3 above)
Comment (yayeoCddy) [reply SRGAN/train_of_srgan.py:185]: ok，我再和复现同事确认下，已改正

**Comment by Ldpe2G** [comment]
image

除了 readme 展示的图片之外，其他的图片都打包然后上传阿里云，让用户下载的方式吧

image
版本号可以去掉
